### PR TITLE
feat: LT-950 - Flags create and manage

### PR DESCRIPTION
**Pull request by cstobart**

### CODE REVIEW

File: flags/serializers.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+class FlagSerializer(serializers.ModelSerializer):
+    team = PrimaryKeyRelatedField(queryset=Team.objects.all())
+    level = serializers.ChoiceField(choices=FlagLevels.choices)
+    status = serializers.ChoiceField(choices=FlagStatuses.choices, default=FlagStatuses.ACTIVE)
+    name = serializers.CharField(
+        max_length=20,
+        validators=[UniqueValidator(queryset=Flag.objects.all(), lookup='iexact',
+                                    message='Enter a name which is not already in use by another flag')],
```
Comment (lukesa) [inline flags/serializers.py]: get strings from content repo

File: flags/serializers.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        error_messages={'blank': 'Flag name may not be blank'})
```
Comment (lukesa) [inline flags/serializers.py]: get strings from content repo

File: flags/views.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+
+@permission_classes((permissions.AllowAny,))
+class FlagsList(APIView):
+    """
+    List all flags
+    """
+    authentication_classes = (GovAuthentication,)
+
+    def get(self, request):
```
Comment (lukesa) [inline flags/views.py]: document each API method

File: flags/views.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        flags = Flag.objects.filter().order_by('name')
+        serializer = FlagSerializer(flags, many=True)
+        return JsonResponse(data={'flags': serializer.data})
+
+    def post(self, request):
```
Comment (lukesa) [inline flags/views.py]: document each API method

File: flags/views.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+
+@permission_classes((permissions.AllowAny,))
+class FlagDetail(APIView):
+    """
+    Details of a specific flag
+    """
+    authentication_classes = (GovAuthentication,)
+
+    def get(self, request, pk):
```
Comment (lukesa) [inline flags/views.py]: document each API method

File: flags/views.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        flag = get_flag(pk)
+        serializer = FlagSerializer(flag)
+        return JsonResponse(data={'flag': serializer.data})
+
+    def put(self, request, pk):
```
Comment (lukesa) [inline flags/views.py]: document each API method

File: flags/views.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        flag = get_flag(pk)
+        data = request.data.copy()
+        if request.user.team != flag.team:
+            return JsonResponse(data={'errors': 'User does not have the rights to change this flag'},
```
Comment (lukesa) [inline flags/views.py]: get strings from content repo

File: flags/serializers.py (hunk 2 above)
Comment (cstobart) [reply flags/serializers.py]: done

File: flags/admin.py [hunk 8]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
+
+# Register your models here.
```
Comment (jfaracik) [inline flags/admin.py]: Remove admin

File: flags/tests/tests_create.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        self.assertEqual(response_data['flag']['name'], 'new flag')
+        self.assertEqual(response_data['flag']['level'], 'Organisation')
+        self.assertEqual(response_data['flag']['team'], str(Team.objects.get(name='Admin').id))
+
+    @parameterized.expand([
+        [{'data': {'name': ''}}],  # Blank
+        [{'data': {'name': 'test'}}],  # Case insensitive duplicate names
+        [{'data': {'name': ' TesT '}}],
+        [{'data': {'name': 'TEST'}}],
+        [{'data': {'name': 'aaaaaaaaaaaaaaaaaaaaaaaaa'}}]  # Too long a name
```
Comment (jfaracik) [inline flags/tests/tests_create.py]: Change to 'a' * number

File: flags/tests/tests_update.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+from teams.models import Team
+from test_helpers.clients import DataTestClient
+
+
+class FlagsUpdateTest(DataTestClient):
+
+    def test_flag_can_be_deactivated(self):
+        flag = Flag(name='New Flag', level='Case', team=self.team)
+        flag.save()
+        data = {
```
Comment (jfaracik) [inline flags/tests/tests_update.py]: spacing around data

File: flags/urls.py [hunk 11]
```diff
@@ -0,0 +1,10 @@
+from django.urls import path
+
+from flags import views
+
+app_name = 'flags'
+
+urlpatterns = [
+    path('', views.FlagsList.as_view(), name='flags'),
```
Comment (jfaracik) [inline flags/urls.py]: Add example url 
eg 
/flags/
/flags/<uuid:pk>/

File: flags/views.py [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+        serializer = FlagSerializer(flags, many=True)
+        return JsonResponse(data={'flags': serializer.data})
+
+    def post(self, request):
+        """
+        Add a new flag
+        """
+        data = JSONParser().parse(request)
+        data['team'] = request.user.team.id
+        data['name'] = data['name'].strip()
```
Comment (jfaracik) [inline flags/views.py]: replace with trim_whitespace in serialiser

File: flags/views.py [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+    def put(self, request, pk):
+        """
+        Edit details of a specific flag
+        """
+        flag = get_flag(pk)
+        data = request.data.copy()
+        if request.user.team != flag.team:
+            return JsonResponse(data={'errors': get_string('flags.error_messages.forbidden')},
+                                status=status.HTTP_403_FORBIDDEN)
+        serializer = FlagSerializer(instance=flag, data=data, partial=True)
```
Comment (jfaracik) [inline flags/views.py]: Spacing

File: flags/views.py [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+        flag = get_flag(pk)
+        serializer = FlagSerializer(flag)
+        return JsonResponse(data={'flag': serializer.data})
+
+    def put(self, request, pk):
+        """
+        Edit details of a specific flag
+        """
+        flag = get_flag(pk)
+        data = request.data.copy()
```
Comment (jfaracik) [inline flags/views.py]: No need for data.copy() unless you're manipulating the data

**Pull request merged**